plugins/trace_api_plugin/abi_data_handler.cpp

```diff
@@ -27,12 +27,16 @@ namespace eosio::trace_api {
                };
                return std::visit([&](auto &&action) -> std::tuple<fc::variant, std::optional<fc::variant>> {
                   using T = std::decay_t<decltype(action)>;
-                  if constexpr (std::is_same_v<T, action_trace_v0>) {
-                     return {serializer_p->binary_to_variant(type_name, action.data, abi_yield), {}};
-                  } else {
-                     return {serializer_p->binary_to_variant(type_name, action.data, abi_yield),
-                             {serializer_p->binary_to_variant(type_name, action.return_value, abi_yield)}};
+                  std::optional<fc::variant> ret_data;
+                  auto params = serializer_p->binary_to_variant(type_name, action.data, abi_yield);
+                  if constexpr (std::is_same_v<T, action_trace_v1>) {
+                     // if there is no return data, an exception will be thrown, so catch it here
+                     try {
+                        ret_data = serializer_p->binary_to_variant(type_name, action.return_value, abi_yield);
+                     }
+                     catch(...) {  }
```

Comment:
This also returns no `ret_data` if `abi_yield` is tripped. I think a cleaner solution is needed here.

Reply:
EPE-1169